### PR TITLE
WIP : add support for new version of cross-val objects

### LINKED ISSUE #2687: Sklearn warnings for cov
I'm running latest `sklearn` and I get this when doing `mne.compute_covariance`:
```
/home/larsoner/.local/lib/python2.7/site-packages/sklearn/cross_validation.py:42: DeprecationWarning: This module has been deprecated in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
/home/larsoner/.local/lib/python2.7/site-packages/sklearn/grid_search.py:43: DeprecationWarning: This module has been deprecated in favor of the model_selection module into which all the refactored classes and functions are moved. This module will be removed in 0.20.
  DeprecationWarning)
```
@agramfort @dengemann any time to look into how to fix this before 0.11 release (in the next week probably), or should we push to 0.12?

### PR DESCRIPTION
**Pull request by agramfort**
try to switch to new sklearn cross-val objects.

I made it work for cv but fore time gen it's way more convoluted.

@kingjr and maybe @dengemann if you have the time to take over this PR to make it work with time gen it would be awesome.

Thank you

closes #2687

### CODE REVIEW

**Comment by kingjr** [comment]
That s going to be some work.... is there a sklearn CV expert around ? I
don't know most CV classes .

**Comment by agramfort** [comment]
>
> That s going to be some work.... is there a sklearn CV expert around ? I
> don't know most CV classes .
>

new CV objects take data info during iterations. For example for LeaveOneOut

you do now:

loo = LeaveOneOut()
for train, test in loo.split(X, y):

so in time_gen you need to pass y and maybe X or labels in the function
that do the split.

clear?

**Comment by agramfort** [comment]
ok merging this.

@kingjr I gave up fixing the time gen code. I don't understand it enough anymore and have no time to dig into it.

**Comment by kingjr** [comment]
>
> > @kingjr <https://github.com/kingjr> I gave up fixing the time gen code.
> I don't understand it enough anymore and have no time to dig into it.
>

What do you suggest ? Shall I make a diagram , have a Skype ?

**Comment by agramfort** [comment]
you see what I did for cov.py and ems.py and you see how it can fly for
time gen :)

File: mne/cov.py [hunk 1]
```diff
@@ -785,7 +785,10 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
                              scalings, n_jobs, stop_early, picks_list,
                              verbose):
     """docstring for _compute_covariance_auto."""
-    from sklearn.grid_search import GridSearchCV
+    try:
+        from sklearn.model_selection import GridSearchCV
+    except: # XXX support sklearn < 0.18
```
Comment (Eric89GXL) [inline mne/cov.py]: boo @agramfort should be `except Exception`. will fix in `master`

Comment (agramfort) [inline mne/cov.py]: arfff ...
